sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -80,7 +80,7 @@ struct codec_info {
 		     bool playback);
 };
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_COMMON_HDMI_CODEC)
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)
```

Comment:
In my opinion, CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC is for HDA codec except HDMI. I am confused CONFIG_SND_HDA_CODEC_HDMI is used in other machine drivers

Reply:
@RanderWang CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC is for all HDA codecs including HDMI. Besides, there is no CONFIG_SND_SOC_SOF_HDA_COMMON_HDMI_CODEC anymore.

Reply:
@RanderWang @bardliao That is correct, SOF_HDA_AUDIO_CODEC is a superset, it is used for all HDA codecs, including HDMI. But note that this only covers the controller side, not the codec driver.

We use CONFIG_SND_HDA_CODEC_HDMI in other machine drivers for this check as that is the direct dependency. E.g. if you have SOF_HDA_AUDIO_CODEC enabled but SND_HDA_CODEC_HDMI is not, you will get build errors from hda_dsp_hdmi_build_controls().  We have made changes to Kconfig rules, so this should not happen anymore, but to avoid issues with randconfig testing, it is better to have explicit checks. So if you have a dependency to SND_HDA_CODEC_HDMI in Kconfig, that should be also used for the #if checks in code.

So I'd prefer these checks to tbe like I did in #1797 (I did this before I realized this PR already existed... I'm ok to merge this instead).